tests/nightly/gpu/test_rag.py

```diff
@@ -388,5 +398,70 @@ def test_bart_fid_rag_dpr_poly(self):
         _test_zoo_file(FID_RAG_DPR_POLY_ZOO_MODEL, True, True)
 
 
+class TestLoadDPRModel(unittest.TestCase):
+    """
+    Test loading different DPR models for RAG.
+
+    See RagAgent._should_override_dpr_model_weights for important note
+    regarding specifying the *same* dpr model file as was used to train
+    the model.
+    """
```

Comment:
Nit: For clarity's sake, might be nice to be explicit about the 3 cases that show up and expected behavior

```
A -> Init DPR Model for M
B -> DPR Model within M after training M
C -> New DPR Model you're using to override
```

Reply:
added a comment!